### DISCUSSION TITLE
Hide items and continue using drag and drop

### DISCUSSION THREAD

**Discussion by Stephan23** [lukasbach/react-complex-tree#348] [Q&A] [answered]
I use an UncontrolledTreeEnvironment with lazy item-loading. 

I would now like to be able to use a switch to change whether deleted items are displayed in the tree or not.
![dnd_deleted](https://github.com/lukasbach/react-complex-tree/assets/912328/2106a429-bfd9-4db6-9266-c72c3aa8531c)

My first approach was to add an `if (item.data.deleted) return <></>` to the renderItem, but this destroys the drop position.

Is there a better way to hide items?

Thanks in advance for your help!

**Comment by Stephan23** [comment]
Duplicate of

[Drag & drop doesn't work properly when an item is hidden](https://github.com/lukasbach/react-complex-tree/issues/112)

but I think the functionality would be a real enrichment.

Best regards

**Comment by lukasbach** [comment, marked as answer]
Are you using a custom tree data provider? You could implement the `getTreeItem` method in a way so that it filters the childrens array before returning to only include items that should be rendered. If something in which items should be rendered and which should not, make sure to call the listeners registered by `onDidChangeTreeData` to notify RCT about which parents now include items whose show/hide behavior has changed, and RCT will update accordingly.
